src/ioctl.rs

```diff
@@ -0,0 +1,29 @@
+//! The IOCTL calls we need for the native linux backend
+
+use nix::{ioctl_read, ioctl_readwrite_buf};
+
+// From linux/hidraw.h
+const HIDRAW_IOC_MAGIC: u8 = b'H';
+const HIDRAW_IOC_GRDESCSIZE: u8 = 0x01;
+const HIDRAW_SET_FEATURE: u8 = 0x06;
+const HIDRAW_GET_FEATURE: u8 = 0x07;
+
+ioctl_read!(
+    hidraw_ioc_grdescsize,
+    HIDRAW_IOC_MAGIC,
+    HIDRAW_IOC_GRDESCSIZE,
+    i32
```

Comment:
probably should use the c_int type instead to make sure this matches with the kernel on all weird archs?

Reply:
I don't know if there are arches where Linux's int isn't 32-bit but it doesn't hurt.